include/zephyr/drivers/video.h

```diff
@@ -202,6 +202,220 @@ struct video_frmival_enum {
+ *
+ * Given a device instance number, a port ID and an endpoint ID, return the endpoint node.
+ * It handles various ways of how a port and an endpoint could be defined as described in
+ * @ref DT_INST_PORT_BY_ID and below. For example:
+ *
+ * @code{.c}
+ *	DT_PROP(DT_INST_ENDPOINT_BY_ID(inst, 0, 0), property)
+ * @endcode
+ *
+ * The above snippet returns @c "endpoint_property" for each of the @c &device listed below:
+ *
+ * @code{.dts}
+ *	&device {
+ *		port {
+ *			endpoint {
+ *				property = "endpoint_property";
+ *			};
+ *		};
+ *	};
+ * @endcode
+ *
+ * @code{.dts}
+ *	&device {
+ *		port {
+ *			endpoint@0 {
+ *				reg = <0x0>;
+ *				property = "endpoint_property";
+ *			};
+ *		};
+ *	};
+ * @endcode
+ *
+ * @code{.dts}
+ *	&device {
+ *		ports {
+ *			port@0 {
+ *				reg = <0x0>;
+ *
+ *				endpoint@0 {
+ *					reg = <0x0>;
+ *					property = "endpoint_property";
+ *				};
+ *			};
+ *		};
+ *	};
+ * @endcode
+ *
+ * @param inst instance number
+ * @param pid port ID
+ * @param eid endpoint ID
+ * @return endpoint node associated with @p eid and @p pid
+ */
+#define DT_INST_ENDPOINT_BY_ID(inst, pid, eid)                                                     \
+	COND_CODE_1(DT_NODE_EXISTS(_DT_INST_ENDPOINT_BY_ID(inst, pid, eid)),                       \
+		(_DT_INST_ENDPOINT_BY_ID(inst, pid, eid)),                                         \
+			(DT_CHILD(DT_INST_PORT_BY_ID(inst, pid), endpoint)))
+
+/**
+ * @brief Get the device node from its endpoint node id.
+ *
+ * Given an endpoint node id, return its device node id.
+ * This handles various ways of how a port and an endpoint could be defined as described in
+ * @ref DT_NODE_BY_ENDPOINT. For example:
+ *
+ * @code{.c}
+ *	DT_PROP(DT_NODE_BY_ENDPOINT(DT_NODELABEL(ep0)), property)
+ * @endcode
+ *
+ * The above snippet returns @c "device_property" in all of the @c ep0 below:
+ *
+ * @code{.dts}
+ *	&device {
+ *		property = "device_property";
+ *
+ *		port {
+ *			ep0: endpoint@0 {
+ *				reg = <0x0>;
+ *			};
+ *		};
+ *	};
+ * @endcode
+ *
+ * @code{.dts}
+ *	&device {
+ *		property = "device_property";
+ *
+ *		ports {
+ *			port@0 {
+ *				reg = <0x0>;
+ *
+ *				ep0: endpoint@0 {
+ *					reg = <0x0>;
+ *				};
+ *			};
+ *		};
+ *	};
+ * @endcode
+ *
+ * @param ep endpoint node
+ * @return device node of @n
+ */
+#define DT_NODE_BY_ENDPOINT(ep)                                                                    \
+	COND_CODE_1(DT_NODE_EXISTS(DT_CHILD(DT_PARENT(DT_GPARENT(ep)), ports)),                    \
+		(DT_PARENT(DT_GPARENT(ep))), (DT_GPARENT(ep)))
+
+/**
+ * @brief Get the remote device object from a local endpoint node id.
+ *
+ * Given an endpoint node id, return the remote device object that this endpoint connects to.
+ * This handles various ways of how a port and an endpoint can be defined as described in
+ * @ref DT_INST_PORT_BY_ID and @ref DT_INST_ENDPOINT_BY_ID.
+ *
+ * @code{.c}
+ *	const struct device *dev1 = DEVICE_DT_GET_REMOTE_DEVICE(DT_NODELABEL(ep0))
```

Comment:
There is no `ep0:` nodelabel in the snippet below. Maybe the example below or this snippet can be updated?

Reply:
yes, it should be `device0_ep_out`, will change.

Reply:
done